tests/integration_poseidon.rs

```diff
@@ -55,113 +69,301 @@ fn test_sanity_check() {
 }
 
 #[test]
-fn test_build_small_tree_par() {
+fn test_build_small_tree() {
     let instruction_path = get_instruction_path();
 
     env_logger::try_init().expect("Invalid logger initialization");
 
     let dclient = DriverClient::new("0", DriverConfig::driver_client_c1100_cfg());
-    let poseidon: PoseidonClient = PoseidonClient::new(Hash::Poseidon, dclient);
-
-    let poseidon = Arc::new(Mutex::new(poseidon));
+    let poseidon: Arc<PoseidonClient> = Arc::new(PoseidonClient::new(Hash::Poseidon, dclient));
 
     let params = PoseidonInitializeParameters {
-        tree_height: 4,
+        tree_height: 3,
         tree_mode: TreeMode::TreeC,
         instruction_path,
     };
+
     let nof_elements = num_of_elements_in_base_layer(params.tree_height);
 
-    let _ = poseidon.lock().unwrap().initialize(params);
+    poseidon.log_api_values();
+
+    poseidon.initialize(params);
+    poseidon.dclient.initialize_cms();
```

Comment:
In MSM we use the name `driver_client` in the MSMClient struct. We should align these terms so the API is familiar between primitives